backend/infrahub/context.py

```diff
@@ -37,3 +44,10 @@ def set_event(self, name: str, id: str) -> None:
             self.event.id = id
         else:
             self.event = EventContext(name=name, id=id)
+
+    def to_event(self) -> dict[str, Any]:
+        return {
+            "branch": self.branch.model_dump(mode="json") if not self.branch.is_global else None,
```

Comment:
I'd suggest to move the logic `model_dump(mode="json") if not self.branch.is_global else None,` into a method of BranchContext.